renku/core/workflow/converters/renku.py

```diff
@@ -0,0 +1,138 @@
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Converter for workflows to Renku workflow description."""
+
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+from renku.core import errors
+from renku.core.plugin import hookimpl
+from renku.core.util import communication
+from renku.core.util.os import get_absolute_path
+from renku.core.util.yaml import write_yaml
+from renku.domain_model.workflow.composite_plan import CompositePlan
+from renku.domain_model.workflow.converters import IWorkflowConverter
+from renku.domain_model.workflow.plan import Plan
+
+
+class RenkuWorkflowFileExporter(IWorkflowConverter):
+    """Export a ``CompositePlan`` or a ``Plan`` to Renku workflow file."""
+
+    @hookimpl
+    def workflow_format(self):
+        """Workflow format name."""
+        return self, ["renku"]
+
+    @hookimpl
+    def workflow_convert(
+        self,
+        workflow: Union[CompositePlan, Plan],
+        basedir: Path,
+        output: Optional[Path],
+        output_format: Optional[str],
+        resolve_paths: Optional[bool],
+        nest_workflows: Optional[bool],
+    ) -> str:
+        """Converts the specified workflow to renku workflow file."""
+
+        def get_output_path():
+            if output:
+                path = output / workflow.name if output.is_dir() else output
+            else:
+                path = Path(tempfile.mkdtemp()) / workflow.name
+
+            absolute_path = Path(get_absolute_path(path))
+            if absolute_path.exists():
+                communication.confirm(f"Output file '{path}' exists. Overwrite it?", abort=True, warning=True)
+
+            return absolute_path
+
+        output_path = get_output_path()
+
+        if isinstance(workflow, CompositePlan):
+            RenkuWorkflowFileExporter.convert_composite_plan(plan=workflow, path=output_path)
+        else:
+            RenkuWorkflowFileExporter.convert_plan(plan=workflow, path=output_path)
+
+        return output_path.read_text()
+
+    @staticmethod
+    def convert_composite_plan(plan: CompositePlan, path: Path):
+        """Converts a composite plan to a workflow file."""
+        data: Dict[str, Any] = {"name": plan.name}
+        if plan.description:
+            data["description"] = plan.description
+        if plan.keywords:
+            data["keywords"] = plan.keywords
```

Comment:
I've added it for the root plan and steps but not for inputs/outputs/parameters since it makes a lot of unnecessary fields. Let me know if you think we should have it for those too.